plutus-benchmark/validation/BenchFull.hs

```diff
@@ -22,36 +21,29 @@ the whole time taken from script deserialization to script execution result.
      `cabal bench -- plutus-benchmark:validation-full --benchmark-options crowdfunding`.
 -}
 main :: IO ()
-main = benchWith mkFullBM
-  where
-    mkFullBM :: FilePath -> BS.ByteString -> Benchmarkable
-    mkFullBM file bsFlat =
-        let
-            body :: Term
-            (UPLC.Program _ v body) = unsafeUnflat file bsFlat
-
-            -- We make some effort to mimic what happens on-chain, including the provision of the
-            -- script arguments. However, the inputs we have are *fully applied*. So we try and
-            -- reverse that by stripping off the arguments here.
-            -- Conveniently, we know that they will be Data constants.
-            -- Annoyingly we can't just assume it's the first 3 arguments, since some
-            -- of them are policy scripts with only 2.
-            (term, args) = peelDataArguments body
-
-            -- strictify and "short" the result cbor to create a real `SerialisedScript`
-            !(benchScript :: SerialisedScript) = force (serialiseUPLC $ UPLC.Program () v term)
-
-        in  whnf (\ script ->
-                      (isRight $ snd $ evaluateScriptRestricting
+main = do
+    evalCtx <- evaluate $ force mkEvalCtx
+    let mkFullBM :: FilePath -> BS.ByteString -> Benchmarkable
+        mkFullBM file bsFlat =
+            let UPLC.Program () ver body = unsafeUnflat file bsFlat
+                -- We make some effort to mimic what happens on-chain, including the provision of
+                -- the script arguments. However, the inputs we have are *fully applied*. So we try
+                -- and reverse that by stripping off the arguments here. Conveniently, we know that
+                -- they will be Data constants. Annoyingly we can't just assume it's the first 3
+                -- arguments, since some of them are policy scripts with only 2.
+                (term, args) = peelDataArguments body
+                -- strictify and "short" the result cbor to create a real `SerialisedScript`
+                !benchScript = force . serialiseUPLC $ UPLC.Program () ver term
+                eval script = do
+                    either (error . show) (\_ -> ()) . snd $ evaluateScriptRestricting
```

Comment:
You can use i think `PlutusPrelude.unsafeFromRight`

Reply:
I think that'd return `()` regardless of whether we get `Left` or `Right` and we do want to throw on `Left` instead.

Reply:
oups yeah